code/osx32_syscalls.cpp

```diff
@@ -40,4 +46,23 @@ extern "C" {
         return (int)syscall3(SYS_mprotect, (void*)addr, (void*)length,
                              (void*)prot);
     }
+
+    void* memset(void *addr, int val, size_t length) {
+        unsigned char *d = (unsigned char*)addr;
+
+        for (; length; length--, d++)
+            *d = val;
+
+        return addr;
+    }
+
+    void* memcpy(void *dest, const void *src, size_t length) {
```

Comment:
This should probably be de-duplicated with https://github.com/fsmv/dfre/blob/master/code/mem_arena.h#L104 (you can also see the style I was using there)

I almost want this memcpy to just call MemCopy (and hopefully that gets inlined). I could maybe be convinced it's better to have everything use the standard memcpy name. I kind of like making it clear it's not the standard lib implementation though.

Reply:
Ehhh, lets just make everything in my code use the standard name (seems it's only used in the platform files). Then it can easily be swapped out later if users want it or for benchmarking.

Just put memset and memcpy in a file called mem_utils.cpp and I think include it at the top of main.cpp before the platform files.

Reply:
You can make my implementation just decrement the NumBytes arg too like your implementation.

Reply:
.

Reply:
I think these should be saved for another branch/PR as well, since I can't test any changes that might affect the other platforms